### PR TITLE
Sign-message improvements

### PR DESCRIPTION
**Pull request by cristure**

### CODE REVIEW

File: api/groups/guardianGroup.go [hunk 1]
```diff
@@ -107,28 +107,21 @@ func (gg *guardianGroup) signMessage(c *gin.Context) {
...
-	signedMsg, otpCodeVerifyData, err := gg.facade.SignMessage(userAddress, userIp, request)
+	signedMsg, otpCodeVerifyData, err := gg.facade.SignMessage(userIp, request)
 	if err != nil {
 		debugErr = fmt.Errorf("%w while signing transaction", err)
 		handleErrorAndReturn(c, getVerifyCodeResponse(otpCodeVerifyData), err.Error())
 		return
 	}
 
-	returnStatus(c, &requests.SignMessageResponse{Message: []byte(request.Message), Signature: signedMsg}, http.StatusOK, "", chainApiShared.ReturnCodeSuccess)
+	returnStatus(c, &requests.SignMessageResponse{Message: request.Message, Signature: string(signedMsg)}, http.StatusOK, "", chainApiShared.ReturnCodeSuccess)
```
Comment (sstanculeanu) [inline api/groups/guardianGroup.go:124]:
```suggestion
	returnStatus(c, &requests.SignMessageResponse{Message: request.Message, Signature: hex.EncodeToString((signedMsg)}, http.StatusOK, "", chainApiShared.ReturnCodeSuccess)
```

File: api/groups/guardianGroup_test.go [hunk 2]
```diff
@@ -848,3 +848,7 @@ func TestHandleHTTPError(t *testing.T) {
 		})
 	}
 }
+
+func TestA(t *testing.T) {
```
Comment (sstanculeanu) [inline api/groups/guardianGroup_test.go:852]: empty test should be removed

File: api/middleware/nativeAuth.go [hunk 3]
```diff
@@ -109,7 +109,7 @@ func (middleware *nativeAuth) MiddlewareHandlerFunc() gin.HandlerFunc {
 			return
 		}
 
-		c.Set(UserAddressKey, string(authToken.GetAddress()))
+		c.Set(UserAddressKey, "erd1qyu5wthldzr8wx5c9ucg8kjagg0jfs53s8nr3zpz3hypefsdd8ssycr6th")
```
Comment (sstanculeanu) [inline api/middleware/nativeAuth.go:112]: this should be reverted to `c.Set(UserAddressKey, string(authToken.GetAddress()))`

also, in order to skip native auth verification on this endpoint, on api.toml, `{ Name = "/sign-message", Open = true, Auth = true },` should have `Auth = false`. This way you basically whitelist this endpoint for having the Bearer mandatory

File: resolver/serviceResolver.go [hunk 4]
```diff
@@ -232,8 +232,8 @@ func (resolver *serviceResolver) VerifyCode(userAddress sdkCore.AddressHandler,
 }
 
 // SignMessage validates user's message, then adds guardian signature and returns the message.
-func (resolver *serviceResolver) SignMessage(userAddress sdkCore.AddressHandler, userIp string, request requests.SignMessage) ([]byte, *requests.OTPCodeVerifyData, error) {
-	guardian, otpCodeVerifyData, err := resolver.verifyCodesReturningGuardian(userAddress, request.GuardianAddr,
+func (resolver *serviceResolver) SignMessage(userIp string, request requests.SignMessage) ([]byte, *requests.OTPCodeVerifyData, error) {
+	guardian, otpCodeVerifyData, err := resolver.verifyCodesReturningGuardian(request.UserAddr, request.GuardianAddr,
```
Comment (sstanculeanu) [inline resolver/serviceResolver.go:236]: here you can create the address like this:

```go
userAddress, err := sdkData.NewAddressFromBech32String(request.UserAddr)
if err != nil {
	return nil, nil, err
}
```

this way, you don't need to further decode the address in `verifyCodesReturningGuardian`, which can keep the same form as before, as it is used from `validateTxRequestReturningGuardian`

File: resolver/serviceResolver.go [hunk 5]
```diff
@@ -413,11 +413,11 @@ func (resolver *serviceResolver) validateTxRequestReturningGuardian(
 		return core.GuardianInfo{}, nil, err
 	}
 
-	return resolver.verifyCodesReturningGuardian(userAddress, txs[0].GuardianAddr, userIp, code, secondCode)
+	return resolver.verifyCodesReturningGuardian(userAddress.AddressAsBech32String(), txs[0].GuardianAddr, userIp, code, secondCode)
 }
 
 func (resolver *serviceResolver) verifyCodesReturningGuardian(
-	userAddress sdkCore.AddressHandler,
+	userAddress string,
```
Comment (sstanculeanu) [inline resolver/serviceResolver.go:420]: not needed, if the above suggestion would be applied

File: resolver/serviceResolver.go [hunk 6]
```diff
@@ -428,16 +428,20 @@ func (resolver *serviceResolver) verifyCodesReturningGuardian(
 		return core.GuardianInfo{}, nil, err
 	}
 
-	addressBytes := userAddress.AddressBytes()
+	addressBytes, err := resolver.pubKeyConverter.Decode(userAddress)
+	if err != nil {
+		return core.GuardianInfo{}, nil, err
+	}
```
Comment (sstanculeanu) [inline resolver/serviceResolver.go:434]: not needed, if the above suggestion would be applied

File: resolver/serviceResolver.go [hunk 7]
```diff
@@ -232,8 +232,8 @@ func (resolver *serviceResolver) VerifyCode(userAddress sdkCore.AddressHandler,
 }
 
 // SignMessage validates user's message, then adds guardian signature and returns the message.
-func (resolver *serviceResolver) SignMessage(userAddress sdkCore.AddressHandler, userIp string, request requests.SignMessage) ([]byte, *requests.OTPCodeVerifyData, error) {
-	guardian, otpCodeVerifyData, err := resolver.verifyCodesReturningGuardian(userAddress, request.GuardianAddr,
+func (resolver *serviceResolver) SignMessage(userIp string, request requests.SignMessage) ([]byte, *requests.OTPCodeVerifyData, error) {
```
Comment (sstanculeanu) [inline resolver/serviceResolver.go:235]: can you add unittests for this method?

File: api/middleware/nativeAuth.go (hunk 3 above)
Comment (cristure) [reply api/middleware/nativeAuth.go:112]: oh wow. I though I didn't commit this one. sorry

File: api/groups/guardianGroup_test.go (hunk 2 above)
Comment (cristure) [reply api/groups/guardianGroup_test.go:852]: this one too

File: resolver/serviceResolver.go (hunk 6 above)
Comment (cristure) [reply resolver/serviceResolver.go:434]: I still need this for the `userInfo` though, right?

File: resolver/serviceResolver.go (hunk 4 above)
Comment (cristure) [reply resolver/serviceResolver.go:236]: done

File: resolver/serviceResolver.go (hunk 5 above)
Comment (cristure) [reply resolver/serviceResolver.go:420]: reverted

File: resolver/serviceResolver.go (hunk 6 above)
Comment (cristure) [reply resolver/serviceResolver.go:434]: reverted

File: resolver/serviceResolver.go (hunk 7 above)
Comment (cristure) [reply resolver/serviceResolver.go:235]: added.

**Review by sstanculeanu — DISMISSED** [review]
tested, works as expected:

input request body:
```json
{
	"code": "850441",
	"second-code":"725944",
	"guardian": "erd1g4e2qs5xzvjdc4tagq34qawt0xvt9mv8u948a05672gsdpnfg46q3k3t0z",
	"message": "my test messageeea",
	"user": "erd1at9keal0jfhamc67ulq4csmchh33eek87yf5hhzcvlw8e5qlx8zq5hjwjl"
}
```

response:
```json
{
	"data": {
		"message": "my test messageeea",
		"signature": "7847d2d111f79f2baba521140be5f1517adacc98e1b2f530650a4dd1486d767dca7662e6e4773faf01c601cd06969c06f9646cb2a3d0fbc6653d26c089952908"
	},
	"error": "",
	"code": "successful"
}
```

File: testscommon/cryptoComponentsHolderStub.go [hunk 8]
```diff
@@ -0,0 +1,54 @@
+package testscommon
+
+import (
+	crypto "github.com/multiversx/mx-chain-crypto-go"
+	"github.com/multiversx/mx-sdk-go/core"
+)
+
+type CryptoComponentsHolderStub struct {
```
Comment (ssd04) [inline testscommon/cryptoComponentsHolderStub.go:8]: add missing mock comment

File: testscommon/signerStub.go [hunk 9]
```diff
@@ -0,0 +1,64 @@
+package testscommon
+
+import (
+	"github.com/multiversx/mx-chain-core-go/data/transaction"
+	crypto "github.com/multiversx/mx-chain-crypto-go"
+)
+
+type SignerStub struct {
```
Comment (ssd04) [inline testscommon/signerStub.go:8]: add missing mock comment

File: testscommon/cryptoComponentsHolderStub.go (hunk 8 above)
Comment (cristure) [reply testscommon/cryptoComponentsHolderStub.go]: added.

File: testscommon/signerStub.go (hunk 9 above)
Comment (cristure) [reply testscommon/signerStub.go]: added.

**Review by sstanculeanu — APPROVED** [review]

**Review by ssd04 — APPROVED** [review]

File: resolver/serviceResolver_test.go [hunk 10]
```diff
@@ -2134,6 +2134,228 @@ func TestServiceResolver_SignTransaction(t *testing.T) {
...
+		args.SignatureVerifier = &testscommon.SignerStub{
+			SignMessageCalled: func(msg []byte, _ crypto.PrivateKey) ([]byte, error) {
+				return []byte("message"), nil
+			},
+		}
+
+		resolver, _ := NewServiceResolver(args)
+
+		assert.NotNil(t, resolver)
+		message, _, err := resolver.SignMessage("userIp", request)
```
Comment (AdoAdoAdo) [inline resolver/serviceResolver_test.go:2353]: the result should be a signature not a message

File: resolver/serviceResolver_test.go [hunk 11]
```diff
@@ -2134,6 +2134,228 @@ func TestServiceResolver_SignTransaction(t *testing.T) {
...
+					GetPrivateKeyCalled: func() crypto.PrivateKey {
+						return nil
+					},
+				}
+				return c, nil
+			},
+		}
+		args.SignatureVerifier = &testscommon.SignerStub{
+			SignMessageCalled: func(msg []byte, _ crypto.PrivateKey) ([]byte, error) {
+				return []byte("message"), nil
```
Comment (AdoAdoAdo) [inline resolver/serviceResolver_test.go:2346]: the result should be a signature not a message

File: resolver/serviceResolver_test.go [hunk 12]
```diff
@@ -2134,6 +2134,228 @@ func TestServiceResolver_SignTransaction(t *testing.T) {
...
+		args.SignatureVerifier = &testscommon.SignerStub{
+			SignMessageCalled: func(msg []byte, _ crypto.PrivateKey) ([]byte, error) {
+				return nil, expectedErr
+			},
+		}
+
+		resolver, _ := NewServiceResolver(args)
+
+		assert.NotNil(t, resolver)
+		txHash, _, err := resolver.SignMessage("userIp", request)
```
Comment (AdoAdoAdo) [inline resolver/serviceResolver_test.go:2311]: signature instead of txHash

**Pull request merged**